scarlet/source.py

```diff
@@ -395,6 +396,104 @@ def __init__(self, frame, sky_coord, observations):
     def _psf_wrapper(self, *parameters):
         return self.frame.psf.__call__(*parameters, bbox=self.bbox)[0]
 
+class StarletSource(FunctionComponent):
+    """Source intialized with starlet coefficients.
+
+    Sources are initialized with the SED of the center pixel,
+    and the morphologies are initialised as ExtendedSources
+    and transformed into starlet coefficients.
+    """
+
+    def __init__(
+        self,
+        frame,
+        sky_coord,
+        observations,
+        obs_idx=0,
+        thresh=1.0,
+        shifting=False,
+    ):
+        """Extended source intialized to match a set of observations
+
+        Parameters
+        ----------
+        frame: `~scarlet.Frame`
+            The frame of the model
+        sky_coord: tuple
+            Center of the source
+        observations: instance or list of `~scarlet.observation.Observation`
+            Observation(s) to initialize this source.
+        obs_idx: int
+            Index of the observation in `observations` to
+            initialize the morphology.
+        thresh: `float`
+            Multiple of the backround RMS used as a
+            flux cutoff for morphology initialization.
+        shifting: `bool`
+            Whether or not a subpixel shift is added as optimization parameter
+        """
+        center = np.array(frame.get_pixel(sky_coord), dtype="float")
+        self.pixel_center = tuple(np.round(center).astype("int"))
+
+        # initialize SED from sky_coord
+        try:
+            iter(observations)
+        except TypeError:
+            observations = [observations]
+
+        # initialize from observation
+        sed, image_morph, bbox = init_extended_source(
+            sky_coord,
+            frame,
+            observations,
+            obs_idx=obs_idx,
+            thresh=thresh,
+            symmetric=True,
+            monotonic=True,
+        )
+
+        sed = Parameter(
+            sed,
+            name="sed",
+            step=partial(relative_step, factor=1e-2),
+            constraint=PositivityConstraint(),
+        )
+
+        noise = mad_wavelet(observations[0].images) * \
+                np.sum(observations[0]._diff_kernels.image**2, axis = (-2,-1))**0.5
```

Comment:
`observations[0]` should be `observations[obs_idx]` to be consistent with the initialization of `image_morph`

Reply:
Also `np.sqrt` is a lot faster than `**0.5`